edisgo/flex_opt/reinforce_measures.py

```diff
@@ -725,3 +732,379 @@ def _replace_by_parallel_standard_lines(lines):
             _replace_by_parallel_standard_lines(relevant_lines.index)
 
     return lines_changes
+
+
+# TODO: check docstrings
+def separate_lv_grid(
+    edisgo_obj: EDisGo, grid: LVGrid
+) -> tuple[dict[Any, Any], dict[str, int]]:
+    """
+    Split LV grid by adding a new substation and connect half of each feeder.
+
+    If the number of overloaded feeders in the LV grid is more than 1 (this can be
+    changed 2 or 3), the feeders are split at their half-length, and the disconnected
+    points are connected to the new MV/LV station.
+
+    1. The point at half the length of the feeders is found.
+    2. The first node following this point is chosen as the point where the new
+       connection will be made. This node can only be a station.
+    3. This node is disconnected from the previous node and connected to a new station.
+    4. New MV/LV is connected to the existing MV/LV station with a line of which length
+       equals the line length between the node at the half-length (node_1_2) and its
+       preceding node.
+
+    Notes:
+    - If the number of overloaded lines in the LV grid is less than 2 (this can be
+      changed 2 or 3) and the node_1_2 is the first node after the main station, the
+      method is not applied.
+    - The name of the new grid will be the existing grid code
+      (e.g. 40000) + 1001 = 400001001
+    - The name of the lines in the new LV grid is the same as the grid where the nodes
+      are removed
+    - Except line names, all the data frames are named based on the new grid name
+
+    Parameters
+    ----------
+    edisgo_obj : :class:`~.EDisGo`
+    grid: class : :class:`~.network.grids.MVGrid` or :class:`~.network.grids.LVGrid`
+
+    Returns
+    -------
+    line_changes=    dict
+        Dictionary with name of lines as keys and the corresponding number of
+        lines added as values.
+    transformer_changes=    dict
+        Dictionary with added and removed transformers in the form::
+        {'added': {'Grid_1': ['transformer_reinforced_1',
+                              ...,
+                              'transformer_reinforced_x'],
+                   'Grid_10': ['transformer_reinforced_10']
+                   }
+        }
+    """
+
+    def get_weight(u, v, data: dict) -> float:
```

Comment:
Parameters u and v are unused. Are they needed?

Reply:
The `get_weight` function is given to the `networkx.algorithms.shortest_paths.weighted._dijkstra` function. This was copied from already existing parts in the code. E.g.:

https://github.com/openego/eDisGo/blob/c6a2edb6db4c55ff81b6ebe9bb216035665d1da3/edisgo/flex_opt/reinforce_measures.py#L412

I didn't check what the dijkstra does with the given function, but I think a simplification creates more overhead than it's worth.

Reply:
The `get_weight` function is given to the `networkx.algorithms.shortest_paths.weighted._dijkstra` function. This was copied from already existing parts in the code. E.g.:

https://github.com/openego/eDisGo/blob/c6a2edb6db4c55ff81b6ebe9bb216035665d1da3/edisgo/flex_opt/reinforce_measures.py#L412

I didn't check what the dijkstra does with the given function, but I think a simplification creates more overhead than it's worth.